src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm.cpp

```diff
@@ -0,0 +1,154 @@
+        CPUKernelExecutor<BrgemmKernelConfig, brgemm_kernel_t>(std::move(kernel_cache), config) {
+    if (config->is_completed())
+        update_kernel();
+}
+
+
+std::shared_ptr<brgemm_kernel_t> BrgemmKernelExecutor::compile_kernel(const std::shared_ptr<BrgemmKernelConfig>& config) const {
+    OV_CPU_JIT_EMITTER_ASSERT(config, "Invalid config provided for BrgemmKernelDesc::compile_kernel");
+    cpu::x64::brgemm_t desc;
+    auto status = brgemm_desc_init(&desc, config->isa, cpu::x64::brgemm_strd,
+                                   config->dt_in0, config->dt_in1,
+                                   false, false, cpu::x64::brgemm_row_major, 1.f,
+                                   config->beta,
+                                   config->LDA, config->LDB, config->LDC,
+                                   config->M, config->N, config->K, nullptr);
+
+    OV_CPU_JIT_EMITTER_ASSERT(status == dnnl_success, "Cannot initialize brgemm descriptor due to invalid params");
+    if (config->is_with_amx) {
+        status = brgemm_init_tiles(desc, config->palette);
+        OV_CPU_JIT_EMITTER_ASSERT(status == dnnl_success, "Cannot initialize brgemm tiles due to invalid params");
+    }
+
+    cpu::x64::brgemm_kernel_t* kernel_ = nullptr;
+    status = brgemm_kernel_create(&kernel_, desc);
+    OV_CPU_JIT_EMITTER_ASSERT(status == dnnl_success, "Cannot create brgemm kernel due to invalid params");
+    return std::unique_ptr<brgemm_kernel_t>(kernel_);
+}
+
+void BrgemmKernelExecutor::execute(const BrgemmKernelExecutor* desc, call_args* args) {
+    const auto& config = desc->m_config;
+    if (config->is_with_amx) {
+        const auto& amx_tile_config = args->amx_tile_config;
+        if (config->M != amx_tile_config->M || config->K != amx_tile_config->K || config->N != amx_tile_config->N) {
+            amx_tile_config->M = config->M;
+            amx_tile_config->K = config->K;
+            amx_tile_config->N = config->N;
+            cpu::x64::amx_tile_configure(config->palette);
+        }
+    }
+
+    cpu::x64::brgemm_kernel_params_t brgemm_p;
+
+    brgemm_p.batch = nullptr;  // default value
+    brgemm_p.ptr_A = args->A;
+    brgemm_p.ptr_B = args->B;
+    brgemm_p.ptr_C = args->C;
+    brgemm_p.ptr_D = args->C;
+    brgemm_p.ptr_buf = args->scratch;
+    brgemm_p.ptr_bias = nullptr;
+    brgemm_p.do_post_ops = static_cast<size_t>(config->is_with_comp);
+    brgemm_p.do_apply_comp = static_cast<size_t>(config->is_with_comp);
+    brgemm_p.skip_accm = 0;
+    brgemm_p.BS = 1;  // default value
+    OV_CPU_JIT_EMITTER_ASSERT(desc->m_kernel, "has nullptr kernel");
+    (*desc->m_kernel)(&brgemm_p);
+}
+
+void BrgemmKernelExecutor::update(size_t M, size_t N, size_t K, size_t LDA, size_t LDB, size_t LDC) {
+    OV_CPU_JIT_EMITTER_ASSERT(m_config, "update is called for empty kernel config");
+#define CAST(X) m_config->X = DIM_CAST(X)
+    CAST(M); CAST(N); CAST(K);
+    CAST(LDA); CAST(LDB); CAST(LDC);
+#undef CAST
+    update_kernel();
```

Comment:
Should we check that `config` is completed?
For example, in dynamic case we get the following sequence of parameters:
1. `M = 256, ...`
2. `M = 1024, ...`
3.  `M = 0, ...`
What will happen in this case (I assume it's possible when tail `work_amount = 0`)? As I got correctly, we will [get](https://github.com/IvanNovoselov/openvino/blob/02913a708524700b7f58e257a64162f7cadab3c0/src/plugins/intel_cpu/src/emitters/snippets/cpu_kernel_executor_table.hpp#L25) exception `Update kernel was called with invalid config`

Reply:
1) We indeed check if config is completed in `update_kernel`, not sure what is the question 🙂 
2 ) M = 0` should not be considered as a valid config in my opinion, because we will emit a lot of useless abi-related code to do nothing. If the brgemm block should be ignored, then the outer loop should have zero work amount.
Also, we discussed that the recompilation is needed only for tail processing (e
.g. you are not supposed to switch block sizes on the fly), and tail brgemm block should always be in the loop.
3) Anyway, if we consider M = 0 a valid config at some point, then we should update the `is_completed()`, and it'll work 🙂

Reply:
Discussed offline.
The main idea: in configurator we will iterate by each `Brgemm` and update their runtime parameters (M, K, N, Leading dimensions). And there can be the situation when some dimension is equal to `0`. For sure, work amount of the corresponding loop will be `0` as well. However, we should have mechanism to skip compilation of these kernels (nothing to compile). At the moment, dnnl [returns](https://github.com/openvinotoolkit/oneDNN/blob/ec7a051b3f4a9e65b22382f3d787e84ce74efe07/src/cpu/x64/brgemm/brgemm.cpp#L175) `invalid_cases` in this case and we will get the [exception](https://github.com/IvanNovoselov/openvino/blob/f19c30224317eec8f4f810f223df62dd46c7a06e/src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm.cpp#L97).
But I'd suspect that `KernelExecutor` can handle these cases and just return `nullptr` kernel. Anyway in execution we will skip the kernel call (since `work_amount` is zero). But we should have mechanism for that. And I think it should be implemented in `Executor`, not in runtime configurator. :blush: